Loop/View Controllers/BolusViewController.swift

```diff
@@ -41,83 +68,284 @@ final class BolusViewController: UITableViewController, IdentifiableClass, UITex
+            case .newCarbEntry, .updatedCarbEntry:
+                title = NSLocalizedString("Meal Bolus", comment: "Title text for bolus screen following a carb entry")
+            }
+        }
+    }
+
+    var originalCarbEntry: StoredCarbEntry? {
+        switch configuration {
+        case .manualCorrection:
+            return nil
+        case .newCarbEntry:
+            return nil
+        case .updatedCarbEntry(from: let entry, to: _):
+            return entry
+        }
+    }
+
+    private var potentialCarbEntry: NewCarbEntry? {
+        switch configuration {
+        case .manualCorrection:
+            return nil
+        case .newCarbEntry(let entry):
+            return entry
+        case .updatedCarbEntry(from: _, to: let entry):
+            return entry
+        }
+    }
+
+    var selectedDefaultAbsorptionTimeEmoji: String?
+
     var glucoseUnit: HKUnit = .milligramsPerDeciliter
 
+    private var computedInitialBolusRecommendation = false
+
     var bolusRecommendation: BolusRecommendation? = nil {
         didSet {
             let amount = bolusRecommendation?.amount ?? 0
             recommendedBolusAmountLabel?.text = bolusUnitsFormatter.string(from: amount)
+
             updateNotice()
-            if let pendingInsulin = bolusRecommendation?.pendingInsulin {
-                self.pendingInsulin = pendingInsulin
+            let wasNoticeRowHidden = oldValue?.notice == nil
+            let isNoticeRowHidden = bolusRecommendation?.notice == nil
+            if wasNoticeRowHidden != isNoticeRowHidden {
+                tableView.reloadRows(at: [IndexPath(row: Row.notice.rawValue, section: 0)], with: .automatic)
+            }
+
+            if computedInitialBolusRecommendation,
+                bolusRecommendation?.amount != oldValue?.amount,
+                bolusAmountTextField.text?.isEmpty == false
+            {
+                bolusAmountTextField.text?.removeAll()
+
+                let alert = UIAlertController(
+                    title: NSLocalizedString("Bolus Recommendation Updated", comment: "Alert title for an updated bolus recommendation"),
+                    message: NSLocalizedString("The bolus recommendation has updated. Please reconfirm the bolus amount.", comment: "Alert message for an updated bolus recommendation"),
+                    preferredStyle: .alert
+                )
+
+                let acknowledgeChange = UIAlertAction(title: NSLocalizedString("OK", comment: "Button text to acknowledge an updated bolus recommendation alert"), style: .default) { _ in }
+                alert.addAction(acknowledgeChange)
+
+                present(alert, animated: true)
             }
         }
     }
 
-    var activeCarbohydratesDescription: String? = nil {
-        didSet {
-            activeCarbohydratesLabel?.text = activeCarbohydratesDescription
+    var maxBolus: Double = 25
```

Comment:
[nit] Shouldn't maximum bolus allowed be from settings?

Reply:
Perhaps optional until gathered from `settings`?

Reply:
Preferable to have some default maximum in case the value isn't set in settings.